### PR TITLE
Fix: teleport Allays with redstone portals 

### LINKED ISSUE #94: Allay teleporting with redstone portals
### Information

* **Server version:**
```
[10:49:15 INFO]: This server is running Paper version git-Paper-491 (MC: 1.20.4) (Implementing API version 1.20.4-R0.1-SNAPSHOT) (Git: c5f68ff)
```

* **Full output of `/mv version -p`:**
https://paste.gg/p/anonymous/0cb944328a9c49538a70717ad4baa245

* **Server log:**
https://mclo.gs/Ed2yLI2

### Details

**Description**  
Allays are not teleported when using [redstone signs](https://github.com/Multiverse/Multiverse-Core/wiki/RedstoneTeleporting-(SignPortals)) set to `ALL`. This is most likely due to Allays not being an instance of [the checked fields](https://github.com/Multiverse/Multiverse-SignPortals/blob/fca61f506cfed33309a04d5e492ad854d78724cc/src/main/java/com/onarandombox/MultiverseSignPortals/utils/PortalDetector.java#L250)

**Steps to reproduce**  
1. Create a signportal that teleports all entities (with the text from the attached screenshot)
2. Summon an Allay (NoAI to make it easier) `/summon allay ~ ~ ~ {NoAI:1b}`
3. Power the signportal and observe the Allay not getting teleported

**Expected behavior**  
The Allay should be teleported

**Screenshots**  
image

### PR DESCRIPTION
**Pull request by zax71**
Fixes #94

### CODE REVIEW

**Comment by dumptruckman** [comment]
Needs a better name for the PR since it will be used for change notes.

File: src/main/java/com/onarandombox/MultiverseSignPortals/utils/PortalDetector.java [hunk 1]
```diff
@@ -227,38 +226,49 @@ public Entity[] getRedstoneTeleportEntities(Sign sign) {
...
-                } else if ((type == ALL || type == MONSTERS) && (entity instanceof Monster)) {
-                    if (entity.getLocation().toVector().isInAABB(min, max)) {
-                        Logging.finest("Found " + entity + " within range!");
-                        entitiesInRange.add(entity);
-                    }
-                } else if ((type == ALL || type == PLAYERS) && (entity instanceof HumanEntity)) {
+                boolean tryToAddEntity = false;
+
+                if (type == RedstoneTeleportType.ALL) {
+                    tryToAddEntity = true;
```
Comment (dumptruckman) [inline src/main/java/com/onarandombox/MultiverseSignPortals/utils/PortalDetector.java:262]: Previously the ALL check was also checking if they were Animals, Squids, Villagers, Monsters, or HumanEntities. Are there no LivingEntities we would not want to include?

Comment (zax71) [reply src/main/java/com/onarandombox/MultiverseSignPortals/utils/PortalDetector.java:262]: I guess display entities and armour stands?

Comment (zax71) [reply src/main/java/com/onarandombox/MultiverseSignPortals/utils/PortalDetector.java:262]: Checking for living entity sounds like a good shout

Comment (zax71) [reply src/main/java/com/onarandombox/MultiverseSignPortals/utils/PortalDetector.java:262]: We already check if the entity is a `LivingEntity` earlier on line 258 by casting, so there is no need to check again

**Review by benwoo1110 — APPROVED** [review]

**Pull request merged**